source/endpoints.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package source
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/labels"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
+func EndpointsForHostname(hostname string, targets endpoint.Targets, ttl endpoint.TTL, providerSpecific endpoint.ProviderSpecific, setIdentifier string, resource string) []*endpoint.Endpoint {
```

Comment:
This exported func is not used. Is this intended ?

Reply:
It is used, is something is showing is not?

![Screenshot 2025-04-24 at 10 01 38](https://github.com/user-attachments/assets/7b18116d-e5c4-4a68-a393-906d2993fd0a)

In follow up pull request I'm happy to use directly in sources. If I do it here, the pull size is going to be even bigger. I already did things that most likely should go in other pull request

Reply:
Sorry, my comment was not clear.

Why exporting a func that is only a used in regular, non-exported func and in tests (of the same package) ? 

I mean, instead of doing:

```go
func Foo(x, y string) {
  /* really do stuff */
}

func foo(x, y string{
  return Foo(x, y)
}

func bar() {
  foo("rand", "om")
}
```

Why not avoid export and reduce code size by doing something like this ?

```go
func foo(x, y string) {
/* really do stuff */
}

func bar() {
  foo("rand", "om")
}
```

Reply:
Agree. Not sure why is needs exporting

Reply:
addressed